src/components/timeline/interfaces.ts

```diff
@@ -0,0 +1,29 @@
+export interface TimelineOptions<T> {
+  items: TimeLineItem<T>[];
+  expanded?: boolean;
+  itemHeaderRenderer: (item?: T) => React.ReactNode | string;
+  itemBodyRenderer?: (item?: T) => React.ReactNode;
+}
+
+export interface TimeLineItem<T> {
+  icon?: React.ReactNode;
+  time: string;
+  value?: T;
+  expanded?: boolean;
+  hasBody: boolean;
+}
+
+export interface UseTimeLineResult<T> {
+  isAllExpanded: boolean;
+  collapseAll: () => void;
+  expandAll: () => void;
+  timelineProps: TimelineProps<T>;
+  Timeline: React.FC<TimelineProps<T>>;
+}
+
+export interface TimelineProps<T> {
```

Comment:
Should this one be able to accept React.HTMLProps so all those props can be passed to the component?

Reply:
Yes why not if we want to customize the component.